Tasks/PublishBuildArtifactsV1/publishbuildartifacts.ts

```diff
@@ -43,13 +44,51 @@ let pathToRobocopyPSString = (filePath: string) => {
     return `'${result}'`;
 }
 
+function createTarArchive(filesPath: string, artifactName: string) {
+    const tar: tr.ToolRunner = tl.tool(tl.which('tar', true));
+    const outputFilePath: string = path.join(tl.getVariable('Agent.TempDirectory'), `${artifactName}.tar`);
+
+    if (tl.stats(filesPath).isFile()) {
+        // If filesPath is a file, we only have to add a single file
+        tar.arg(['cf', outputFilePath, '--directory', path.dirname(filesPath), path.basename(filesPath)]);
+    } else {
+        // If filesPath is a directory, we have to add all files from that directory to the tar archive
+        tar.arg(['cf', outputFilePath, '--directory', filesPath, '.']);
+    }
+
+    const tarExecResult: tr.IExecSyncResult = tar.execSync();
+
+    if (tarExecResult.error || tarExecResult.code !== 0) {
+        throw new Error(`Couldn't add artifact files to a tar archive: ${tarExecResult.error}`);
+    }
+
+    return outputFilePath;
+}
+
+function getPathToUpload(
```

Comment:
Could it make sense to rename this method? Currently it doesn't only return upload path - for some cases it creates archive, which could be unobvious

Reply:
Yeah, renamed it